extension/src/test/XLIFFTypes.test.ts

```diff
@@ -635,6 +639,70 @@ suite("Xliff Types - Functions", function () {
       "Unexpected contents of array"
     );
   });
+
+  test("TransUnit.setTargetStateFromToken", function () {
+    const tu = TransUnit.fromString(getTransUnitXml());
```

Comment:
Hmmm... The trans-unit in getTransUnitXml() already has States, not Tokens... So setTargetStateFromToken() wouldn't do anything at all here, right? But maybe that's a test as well... might want a comment that describes what is being tested. Or maybe this test should be split into several tests?

I mean, a test should most often consist of 
one or more GIVEN
one WHEN
one or more THEN

This is more like GIVEN, WHEN, THEN, THEN, THEN, WHEN, THEN, THEN, THEN, WHEN, THEN, THEN, etc 😁


I know we do have other test functions that contains a lot of similar tests in one function, but they should at least not be dependent of each other

Reply:
Yeah, the first part was meant mostly as a sanity check in case the string was changed. The lazy workings of a tired mind I guess :sweat_smile: 

I've reworked this to use a new instance for each test. Please note the new return signature of `setTargetFromToken()` - Keep or scrap? 

I do think there's a lot to say and do about our current test structure. It would be a lot of work to re-arrange it but if we could find a nice, easy to follow and scalable structure it could be worth it.

Reply:
I don't know if it's worth the effort to rebuild all tests "just because". But we can always improve as we are moving forward, and then we can rework the old tests when we need to touch them, don't you think?

Reply:
Absolutely, it's just a bit messy in some places at the moment :smile:

Reply:
Agreed!